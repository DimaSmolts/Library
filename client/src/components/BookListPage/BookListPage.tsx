import React from 'react'
import BookListContainer from '../BookList/BookListContainer'
import SearchPanelContainer, { SearchPanelProps } from '../SearchPanel/SearchPanelContainer'

export default class BookListPage extends React.Component<SearchPanelProps> {
  render() {
    return <>
      <SearchPanelContainer queryParams={this.props.queryParams} history={this.props.history} />
      <BookListContainer />
    </>
  }
}
