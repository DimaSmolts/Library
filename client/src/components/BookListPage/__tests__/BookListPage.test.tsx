import React from 'react'
import { shallow } from 'enzyme'
import { createBrowserHistory } from 'history'

import BookListPage from '../BookListPage'

const queryParams = 'anything'
const history = createBrowserHistory()

test('snapshot: <BookListPage /> renders correctly', () => {
  const tree = shallow(
    <BookListPage queryParams={queryParams} history={history} />)

  expect(tree).toMatchSnapshot()
})
