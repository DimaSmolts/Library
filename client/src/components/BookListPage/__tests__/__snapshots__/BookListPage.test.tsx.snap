// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: <BookListPage /> renders correctly 1`] = `
<Fragment>
  <Connect(SearchPanelContainer)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    queryParams="anything"
  />
  <Connect(BooksListContainer) />
</Fragment>
`;
